### ISSUE TITLE
Ping broken after a while

### ISSUE THREAD

**Issue by daywalker90** [ElementsProject/lightning#7742]
I wrote a plugin to ping my channel peers every hour and i noticed that after a while every node hit my timeout of 4s. I tried to run the ping command from the cli but it would never return. I also don't see anything related in the logs. The ping command seems to be stuck for my whole node. Reconnecting the peer alone does not help. Only after restarting the node the ping command works again.

**Comment by rustyrussell** [comment]
Yes, the ping command is unreliable.  We are only supposed to ever send one ping at a time, but we also send them internally, so the manual ping can get confused, thinking it has never received a reply.  I will fix!

**Comment by rustyrussell** [comment]
Actually, it does look like they should piggyback on our keep alive pings?  I'll see if I can reproduce this with aggressive testing...

**Comment by rustyrussell** [comment]
I cannot reproduce this, at least in master.

You *will* get a ping failure, in that it will report "'error': {'code': -1, 'message': 'Ping already pending'}}" if you happen to hit the time when we're sending a ping anyway.  For your "keepalive" use case, you should ignore these failures.

**Comment by wtogami** [comment]
His plugin was not for keepalive. Rather it was for measuring latency. It is fine for it to ignore temporary errors. But the problem he ran into earlier was after a while all pings would stop working until the node was restarted.

**Comment by daywalker90** [comment]
I tested pinging again on mainnet with v25.02 and quickly got into the problem again.

I then tried master and indeed i could not get it stuck like that! But:

The ping performance seems to quickly degrade alot!

I attached a file where i connect to 10 peers at a time, ping them and then disconnect. I also do a simple tcp handshake for comparison. I attached a file with the results. The timeout for both ping and handshake is 30s. If it says tcp: false the handshake failed. Rest should be obvious. I did not include tor only nodes.

[result.txt](https://github.com/user-attachments/files/20272512/result.txt)

**Comment by wtogami** [comment]
#8308 contains specific details that appear to be related to why we're seeing ping unreliability.
